README.md

```diff
@@ -1,102 +1,190 @@
- - `pathPattern` : Regular expressions that will be tested on the node path.
- - `workspace` : `live` or `default`, only request on the specified workspace will match.
- - `nodeType` : Only request on nodes of these type will match.
- - `scope` : Only request claiming that scope will match. 
- - `permission` : Only request with users having this permission will match.
- 
-Note the difference between `requiredPermission`/`requiredScope` and `permission`/`scope` - in the "non-required" case, if the user has the permission/scope, his access will be granted (based on `access`, `requiredPermission`, `requiredScope` values), and no other rule will be checked. 
-But it's not required - if he does not have it, the rule won't match, and so subsequent rules will be checked. 
-In the "required" case, if the rule match, the user must have the permission/scope - no other rule will be checked. 
-
-An optional `priority` property can be also specified - the rules with the lowest value will be executed first. Default priority is 0.
-
-```
-permission.checkFirst.priority=-10
+```yaml
+graphql:
+  description: Can access graphql API
+  metadata:
+    visible: true
+  auto_apply:
+    - origin: hosted
+  grants:
+    - api: graphql
+      node: none
+    - api: graphql
+      node:
+        withPermission: api-access
 ```
 
-A rule with no condition will always match (and thus, should have a very high value for priority so that it is checked last):
-```
-permission.global.access=restricted
-permission.global.priority=99999
+will be written this way when using cfg format : 
+```properties
+graphql.description = Can access graphql API
+graphql.metadata.visible = true
+graphql.auto_apply[0].origin = hosted
+graphql.grants[0].api = graphql
+graphql.grants[0].node = none
+graphql.grants[1].api = graphql
+graphql.grants[1].node.withPermission = api-access
 ```
 
-Rules can specify a simple path pattern : 
-```
-permission.users.pathPattern=/sites/[^/]+/home/.*
+Examples below are given in yaml format.
+
+### Scope name, description and metadata
+
+Every scope must have a unique name. The description explains what the scope is granting. 
+Metadata can be freely added and used by UI or other services.
+
+### Scope grants
+
+A scope contains a list of grants, one for each API access. 
+
+- `api` : The names of the API (in a comma separated list), if the rule should only apply to some entry points.
+  Different API names are provided by the API services. For example : ajax views (`view.<view-type>`), the JCRest API module (`jcrestapi`), or the GraphQL API (`graphql.<gql-type>.<gql-field>`)
+  
+    ```yaml
+       grants: 
+         - api: graphql.JcrNode, graphql.JcrProperty
+    ```
+
+    You can also define `include` and `exclude` sub entries :
+
+    ```yaml
+       grants:
+         - api:
+             include: graphql
+             exclude: graphql.GqlAdmin, graphql.JcrNode
+    ```
+    
```

Comment:
Suggested addition (if that helps):
"Remember that scopes are only granting access, when using the "exclude" key you are removing the corresponding node from the element to be granted but you are not denying access to that element.